### ISSUE TITLE
Segmentation fault when starting SLI in NEST 2.16.0

### ISSUE THREAD

**Issue by janhahne** [nest/nest-simulator#1013]
I just downloaded the NEST 2.16.0 release from github ( https://github.com/nest/nest-simulator/releases/download/v2.16.0/nest-2.16.0.tar.gz ) and got an segmentation fault when I started "nest" (the SLI Version)

```
hahne@hahne-ubuntu:~$ nest
[INFO] [2018.8.22 9:18:9 /home/hahne/git/nest-2.16.0/nestkernel/rng_manager.cpp:238 @ Network::create_rngs_] : Creating default RNGs
[INFO] [2018.8.22 9:18:9 /home/hahne/git/nest-2.16.0/nestkernel/rng_manager.cpp:284 @ Network::create_grng_] : Creating new default global RNG
Speicherzugriffsfehler (Speicherabzug geschrieben)
``` 

Probably for the same reason "make installcheck" crashes during the first test with an segmentation fault. I experienced this issue under Ubuntu 18.04 with gcc 7.3 and was also able to reproduce it under OpenSuse 13.2 with gcc  4.8.1.

**Comment by stinebuu** [comment]
@janhahne I can reproduce this issue. When I try to start NEST I get an error message saying

```
Aug 22 11:44:17 SLIStartup [Fatal]: 
    Your NEST installation seems broken. 


Aug 22 11:44:17 SLIStartup [Fatal]: 
    I could not find the startup file that

Aug 22 11:44:17 SLIStartup [Fatal]: 
    should have been in 
    /home/terhorst/workflows/nest-release-tools/install/share/nest
```

So it seems like nest looks for the startup file in @terhorstd folder on my computer? Which will not work of course.

**Comment by hakonsbm** [comment]
@janhahne @stinebuu @terhorstd Looks like the release contains some files that should not be there. Specifically it contains `libnestutil/config.h`, which is supposed to be generated by CMake. Removing it and letting CMake generate a new one solves the issue for me.

In addition to `config.h`, it also contains other generated files, like `setup.py`, `nest-config`, `nest-vars.sh`, generated CMake files, `do_tests.sh`, and some logfiles.

**Comment by heplesser** [comment]
@hakonsbm @terhorstd @jougs  Could it be that the 2.16.0 tarball includes files it shouldn't, while checking out the v2.16.0 tag using git would give one a clean source tree without any of the cmake-generated files?

@janhahne @stinebuu You seem to encounter two rather different types of error, one a segfault, the other a "file not found".

**Comment by janhahne** [comment]
I think @hakonsbm analysis is correct. Removing `libnestutil/config.h` beforehand solved the issue for me as well.

**Comment by heplesser** [comment]
I now downloaded all three archives offered at 
https://github.com/nest/nest-simulator/releases/tag/v2.16.0. Only the `nest-2.16.0.tar.gz` file contains `config.h` and other spurious files, the other two archives are identical and fine. 

@jougs @terhorstd Could you update the `nest-2.16.0.tar.gz` file? I think it should keep the revision number, since it is only a mispacked archive. Or do we need to go to 2.16.1 to avoid confusion?

**Comment by janhahne** [comment]
@heplesser @jougs @terhorstd The Zenodo entry http://doi.org/10.5281/zenodo.1400175 unfortunately also contains the archive with the additional files.

**Comment by heplesser** [comment]
Since Zenodo presumably is permanent, I assume we need to creare 2.16.1 immediately. It should have the same authors as 2.16.0. 

Why do we need the extra tarball at all, in addition to the ones Github provides automatically? I think we discussed this once, but I forgot again.

**Comment by jougs** [comment]
During the creation of the custom release tarball (`nest-2.16.tar.gz`; on a machine by @terhorstd, thus the references to his home dcirectory), we executed `cmake` in the source directory in order to acquire a `Makefile` that permits us to run `make dist`. Apparently that run both created and packed the build artifacts that now cause the trouble.

The reason for having our own tarball in the first place is that historically, we were doing additional things during the `make dist` step and that the tarball created by GitHub and the directory it unpacks to is named differently from what we were used to from the `autotools` era.

This accident (and the additional effort for creating the custom tarball) makes we wonder if we should rather update our installation instructions to use the GitHub names and abandon our own tarball altogether.

Any opinions?

**Comment by jougs** [comment]
An while we're on it, saving the world: @janhahne: can you please switch your OS to English?! The word `Speicherabzug` makes me cry ;-)

**Comment by heplesser** [comment]
@jougs I agree that we should stick to English, but thanks to @janhahne for finally enlightening me to what "core dump" is in my native language ;).

**Comment by heplesser** [comment]
@jougs I agree with your suggestion: Why should be do things manually that Github offers us automatically, and confuse the world with two minimally different tarballs. So dropping the manual tarball and updating our instructions to `nest-simulator-....` makes sense. Would you create an issue for that? The only thing we would need to check before is whether linking to the Github-autogenerated archive from Zenodo works properly. @terhorstd Could you look at that?

**Comment by jougs** [comment]
@heplesser: just in case this was not clear: we never linked to a tarball on GitHub from Zenodo, but rather uploaded one there. This will definitely also work in the future. But having just the release without additional files, we can also go a step further and just [cite the release](https://guides.github.com/activities/citable-code).
